dockerfiles/che/entrypoint.sh

```diff
@@ -324,14 +324,60 @@ add_che_cert_to_truststore() {
 }
 
 add_public_cert_to_truststore() {
+  JAVA_TRUST_STORE=/home/user/cacerts
+  DEFAULT_JAVA_TRUST_STOREPASS="changeit"
+
+  chmod 644 $JAVA_TRUST_STORE || true
+
   CUSTOM_PUBLIC_CERTIFICATES="/public-certs"
   if [[ -d "$CUSTOM_PUBLIC_CERTIFICATES" && -n "$(find $CUSTOM_PUBLIC_CERTIFICATES -type f)" ]]; then
     FILES="$CUSTOM_PUBLIC_CERTIFICATES/*"
     for cert in $FILES
     do
-      add_cert_to_truststore "$(<$cert)" "HOSTDOMAIN-$(basename $cert)"
+      jks_import_ca_bundle $cert $JAVA_TRUST_STORE $DEFAULT_JAVA_TRUST_STOREPASS
     done
   fi
+
+  chmod 444 $JAVA_TRUST_STORE
```

Comment:
Also when you're done making changes to the trust store, it's a good idea to make the file uneditable by the `user` user again, so that no more certs can be imported in that container. But on Openshift, this might fail if your user doesn't have permission, so it's better to NOT fail here:

```
  chmod 444 $JAVA_TRUST_STORE || true
```
See https://github.com/eclipse/che/pull/17563/files#diff-0494d7f08e22baf0f7a6a136eb6f5a6a6532d3199d4edda56964dae41c18b22bR310-R311

Reply:
Agree, will fix

Reply:
According to the suggested solution `|| true` shouldn't be needed any more.

Reply:
unless your rhel.dockerfile grants `777` permission to the file, this will likely fail.